For the reference, here are failures from two plot tests when all dependencies are added:

```
__________________________ test_umap_plot_testability __________________________

    def test_umap_plot_testability():
        try:
>           from umap import plot

umap/tests/test_plot.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
umap/plot.py:7: in <module>
    import datashader as ds
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/__init__.py:8: in <module>
    from .core import Canvas                                 # noqa (API import)
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/core.py:19: in <module>
    from . import reductions as rd
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/reductions.py:11: in <module>
    from datashader.transfer_functions._cuda_utils import (cuda_atomic_nanmin,
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/transfer_functions/__init__.py:18: in <module>
    from datashader.composite import composite_op_lookup, over
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/composite.py:17: in <module>
    def extract_scaled(x):
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/decorators.py:211: in wrapper
    disp.enable_caching()
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/dispatcher.py:743: in enable_caching
    self._cache = FunctionCache(self.py_func)
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/caching.py:620: in __init__
    self._impl = self._impl_class(py_func)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <numba.core.caching.CompileResultCacheImpl object at 0x7fff9dde1b80>
py_func = <function extract_scaled at 0x7fff9ddecf70>

    def __init__(self, py_func):
        self._lineno = py_func.__code__.co_firstlineno
        # Get qualname
        try:
            qualname = py_func.__qualname__
        except AttributeError:
            qualname = py_func.__name__
        # Find a locator
        source_path = inspect.getfile(py_func)
        for cls in self._locator_classes:
            locator = cls.from_function(py_func, source_path)
            if locator is not None:
                break
        else:
>           raise RuntimeError("cannot cache function %r: no locator available "
                               "for file %r" % (qualname, source_path))
E           RuntimeError: cannot cache function 'extract_scaled': no locator available for file '/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/composite.py'

/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/caching.py:355: RuntimeError
____________________________ test_plot_runs_at_all _____________________________

mapper = UMAP(n_epochs=100)
iris = {'data': array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5,...r66865jka1d5jcvczmlk1k4hknb4-python3.8-scikit-learn-0.23.2/lib/python3.8/site-packages/sklearn/datasets/data/iris.csv'}

    def test_plot_runs_at_all(mapper, iris):
>       from umap import plot as umap_plot

umap/tests/test_plot.py:39:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
umap/plot.py:7: in <module>
    import datashader as ds
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/__init__.py:8: in <module>
    from .core import Canvas                                 # noqa (API import)
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/core.py:19: in <module>
    from . import reductions as rd
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/reductions.py:11: in <module>
    from datashader.transfer_functions._cuda_utils import (cuda_atomic_nanmin,
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/transfer_functions/__init__.py:18: in <module>
    from datashader.composite import composite_op_lookup, over
/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/composite.py:17: in <module>
    def extract_scaled(x):
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/decorators.py:211: in wrapper
    disp.enable_caching()
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/dispatcher.py:743: in enable_caching
    self._cache = FunctionCache(self.py_func)
/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/caching.py:620: in __init__
    self._impl = self._impl_class(py_func)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <numba.core.caching.CompileResultCacheImpl object at 0x7fff8d3be730>
py_func = <function extract_scaled at 0x7fff8d827430>

    def __init__(self, py_func):
        self._lineno = py_func.__code__.co_firstlineno
        # Get qualname
        try:
            qualname = py_func.__qualname__
        except AttributeError:
            qualname = py_func.__name__
        # Find a locator
        source_path = inspect.getfile(py_func)
        for cls in self._locator_classes:
            locator = cls.from_function(py_func, source_path)
            if locator is not None:
                break
        else:
>           raise RuntimeError("cannot cache function %r: no locator available "
                               "for file %r" % (qualname, source_path))
E           RuntimeError: cannot cache function 'extract_scaled': no locator available for file '/nix/store/fbc35lajrwk092scbknm79ljvgga493m-python3.8-datashader-0.11.1/lib/python3.8/site-packages/datashader/composite.py'

/nix/store/j507gq4jndxp1gbf489k1rwqzdy9qx2z-python3.8-numba-0.51.1/lib/python3.8/site-packages/numba/core/caching.py:355: RuntimeError

```

--- a/pkgs/development/python-modules/umap-learn/default.nix
+++ b/pkgs/development/python-modules/umap-learn/default.nix
@@ -6,21 +6,23 @@
 , scikitlearn
 , scipy
 , numba
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "umap-learn";
-  version = "0.3.10";
+  version = "0.4.5";
 
   src = fetchFromGitHub {
     owner = "lmcinnes";
     repo = "umap";
     rev = version;
-    sha256 = "0nck5va5km7qkbrhn15dsn0p2mms9kc641lcypy7l8haqgm44h8x";
+    sha256 = "080by8h4rxr5ijx8vp8kn952chiqz029j26c04k4js4g9s7201bq";
   };
 
   checkInputs = [
     nose
+    pytestCheckHook
   ];
 
   propagatedBuildInputs = [
@@ -30,15 +32,15 @@ buildPythonPackage rec {
     numba
   ];
 
-  postConfigure = ''
-   substituteInPlace umap/tests/test_umap.py \
-     --replace "def test_umap_transform_on_iris()" "@SkipTest
-def test_umap_transform_on_iris()"
-  '';
+  disabledTests = [
+    # Plot functionality requires additional packages.
+    # These test also fail with 'RuntimeError: cannot cache function' error.
+    "test_umap_plot_testability"
+    "test_plot_runs_at_all"
 
-  checkPhase = ''
-    nosetests -s umap
-  '';
+    # Flaky test. Fails with AssertionError sometimes.
+    "test_sparse_hellinger"
+  ];
 
   meta = with lib; {
     description = "Uniform Manifold Approximation and Projection";

